cpp/oneapi/dal/table/backend/csr_kernels.cpp

```diff
@@ -0,0 +1,238 @@
+
+    pull_column_indices_impl<Policy, BlockData>(policy,
+                                                origin_column_indices,
+                                                origin_offset,
+                                                block_size,
+                                                block,
+                                                kind,
+                                                preserve_mutability);
+
+    pull_row_indices_impl<Policy, BlockData>(policy,
+                                             origin_row_indices,
+                                             block_info,
+                                             block,
+                                             kind,
+                                             preserve_mutability);
+}
+
+template <typename Policy, typename BlockData>
+void csr_pull_block(const Policy& policy,
+                    const csr_info& origin_info,
+                    const block_info& block_info,
+                    const array<byte_t>& origin_data,
+                    const array<std::int64_t>& origin_column_indices,
+                    const array<std::int64_t>& origin_row_indices,
+                    detail::csr_block<BlockData>& block,
+                    alloc_kind requested_alloc_kind,
+                    bool preserve_mutability) {
+    switch (origin_info.layout_) {
+        case data_layout::row_major:
+            pull_csr_block_impl(policy,
+                                origin_info,
+                                block_info,
+                                origin_data,
+                                origin_column_indices,
+                                origin_row_indices,
+                                block,
+                                requested_alloc_kind,
+                                preserve_mutability);
+            break;
+        default: throw dal::domain_error(error_msg::unsupported_data_layout());
+    }
+}
+
+#define INSTANTIATE(Policy, BlockData)                                             \
+    template void csr_pull_block(const Policy& policy,                             \
+                                 const csr_info& origin_info,                      \
+                                 const block_info& block_info,                     \
+                                 const array<byte_t>& origin_data,                 \
+                                 const array<std::int64_t>& origin_column_indices, \
+                                 const array<std::int64_t>& origin_row_indices,    \
+                                 detail::csr_block<BlockData>& block,              \
+                                 alloc_kind requested_alloc_kind,                  \
+                                 bool preserve_mutability);
+
+#ifdef ONEDAL_DATA_PARALLEL
+#define INSTANTIATE_ALL_POLICIES(Data)             \
+    INSTANTIATE(detail::default_host_policy, Data) \
+    INSTANTIATE(detail::data_parallel_policy, Data)
```

Comment:
How this is instantiated in case of data parallel policy?

Reply:
In the case of DPC++ policy it should throw an exception.

Reply:
It was fixed